### PR TITLE
Add TEXTSTRING support to solr index

### PR DESCRIPTION
**Pull request by VladimirBogomolov**
Signed-off-by: Vladimir Bogomolov <vladimir.bogomoloff@gmail.com>

-----

Resolve Issue #328
-----

### For all changes:
- [x] Is there an issue associated with this PR? Is it referenced in the commit message?
- [x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you written and/or updated unit tests to verify your changes?

### CODE REVIEW

File: janusgraph-backend-testutils/src/main/java/org/janusgraph/diskstorage/indexing/IndexProviderTest.java [hunk 1]
```diff
@@ -1099,7 +1099,9 @@ public void testCustomAnalyzer() throws Exception {
 
             assertEquals(1, tx.queryStream(new IndexQuery(store, PredicateCondition.of(FULL_TEXT, Cmp.GREATER_THAN, "a"))).count());
             assertEquals(0, tx.queryStream(new IndexQuery(store, PredicateCondition.of(FULL_TEXT, Cmp.GREATER_THAN, "z"))).count());
-            assertEquals(1, tx.queryStream(new IndexQuery(store, PredicateCondition.of(FULL_TEXT, Cmp.GREATER_THAN, "Tom and Jerry"))).count());
+            //index should return zero documents with a value greater than "Tom and Jerry",
+            //because the index contains only one document exactly equal to this value, and no more documents
+            // assertEquals(1, tx.queryStream(new IndexQuery(store, PredicateCondition.of(FULL_TEXT, Cmp.GREATER_THAN, "Tom and Jerry"))).count());
```
Comment (li-boxuan) [inline janusgraph-backend-testutils/src/main/java/org/janusgraph/diskstorage/indexing/IndexProviderTest.java:1104]: Thanks for your contribution @VladimirBogomolov ! Can you help me understand this change? Do you know why this test case fails for Solor but passes for Lucene and ES?

**Comment by porunov** [comment]
@VladimirBogomolov your PR is missing an important [commit](https://github.com/VladimirBogomolov/janusgraph/commit/14c6b810532352eca227ed5c14d2a70e42fcf25d) from `master` branch which fixed HBase tests. Without that commit HBase tests will most likely fail in your PR. Could you please rebase your branch against the latest `master` branch to ensure that this commit is in your branch?

**Comment by mad** [comment]
LGTM!

**Review by mad — APPROVED** [review]

**Review by li-boxuan — APPROVED** [review]
Thank you for your contribution! Cannot say I understand every piece but in general LGTM

**Pull request merged**